### DISCUSSION TITLE
Nominatim 4.1 Discussion

### DISCUSSION THREAD

**Discussion by philipkozeny** [mediagis/nominatim-docker#356] [General]
Here are some things off the top of my head we should discuss before we start working on the 4.1 upgrade:

- choose the proper base distribution
    - switch to Ubuntu 22.04 LTS from 20.04 LTS: should be done if we stay on Ubuntu
    - switch to Alpine: we need to discuss the benefits & check if all 3rd party packages are supported
    - pro: official documentation from upstream for 22.04
- consider separating the DB from the App 
    - with the new ICU tokenizer, this would be doable
    - not be a sensible default since the majority like the one-command-one-go approach a lot
- upgrade path
    - I would advise against providing an official upgrade path from our older versions since there are changes in Nominatim, PostgreSQL, etc. that all have to be considered and thoroughly tested

Anything else I have missed or you want to discuss?

**Comment by philipkozeny** [comment]
@leonardehrenfried any thoughts from your side?

**Comment by leonardehrenfried** [comment]
Sorry, I typed up a response but apparently never sent it. Since I won't be doing the upgrade this time round, here are my thoughts. Since I want to keep things simple for the maintainers I would do the following:

- Stay on Ubuntu but upgrade to 22.04. Switching to another base image would probably mean more testing and refactoring which I would not want to do since I don't see any compelling advantages of switching.
- To me having a single container for everything was the main selling point and I would not want to split the app from the DB. Anyway, if you want to do that it's now possible with a bit of configuration.
- Having an supported upgrade path is bound to lead to lots of support requests and I don't have the bandwidth to deal with it so I'm also against it.
